gateway/reverse_proxy.go

```diff
@@ -1349,6 +1367,16 @@ func (p *ReverseProxy) WrappedServeHTTP(rw http.ResponseWriter, req *http.Reques
 
 	p.TykAPISpec.Lock()
 
+	isTimeoutEnforced, enforcedTimeout := checkHardTimeoutEnforced(p.TykAPISpec, outreq)
+
+	// limit request time with context timeout
+	if isTimeoutEnforced {
```

Comment:
instead of this what if we call `if spec.EnforcedTimeoutEnabled {` and inside of `checkHardTimeoutEnforced` we just do

```
vInfo, _ := spec.Version(req)
	versionPaths := spec.RxPaths[vInfo.Name]
	found, meta := spec.CheckSpecMatchesStatus(req, versionPaths, HardTimeout)
	if found {
		intMeta := meta.(*int)
		p.logger.Debug("HARD TIMEOUT ENFORCED: ", *intMeta)
		return true, float64(*intMeta)
	}
```

Reply:
Seems subjective. What's the benefit?

Reply:
Reducing two if check to one

Reply:
it's not even the same value, if no hard timeout or invalid timeout is configured it will return false explicitly even if .EnforcedTimeoutEnabled is true

Reply:
I'm rejecting this suggestion, as it just isn't good.

WrappedServeHTTP is SLOC a billion, CheckHardTimeouts func is SLOC ~20.
Function currently has locality of behaviour, splitting the condition out wouldn't.
Early returns are best, as they are easiest to understand for the reader.

It's a bad api to begin with, as the logger is passed via *ReverseProxy, the function is attached to the wrong scope or rather should get the logger explicitly and not even be aware of *ReverseProxy.